File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 9/8/2025, 4:38:42 PM
class Solution {
    public boolean isPalindrome(int x) {
        String y = ""+x;
        String reverse ="";
        for(int i = y.length()-1; i>=0; i--){
            reverse+=y.charAt(i);
        }
        return reverse.equals(y);
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 2/15/2026, 8:54:26 PM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            string = str(x)
            l = 0
            r = len(string)-1
            while(l < r):
                if (string[l] != string[r]):
                    return False
                l += 1
                r -= 1         
            return True